db/migrate/20170221141514_create_notification_log.rb

```diff
@@ -0,0 +1,19 @@
+class CreateNotificationLog < ActiveRecord::Migration
+  def change
+    create_table :notification_logs do |t|
```

Comment:
@tijmenb @MatMoore I was avoiding adding validations on this model as:

* I don't want the process to fail if this model fails to save. A logging failure should not trigger an email to be resent (i.e. don't raise an error).
* I would prefer to save all the records (including invalid ones) so that I can understand what missing/invalid data is occurring and identify issues in the process. Once we have a better view of the data I would potentially look to add some validation at that stage.